### ISSUE TITLE
Run PyTorch testing suite against syft.interfaces.torch

### ISSUE THREAD

**Issue by jeremyjordan** [OpenMined/OpenMined#286]
## Context
We’d like to develop OpenMined such that the functionality mirrors what you could do on PyTorch, but you have more flexibility on where you train your models (thanks to Unity). As a developer, I’d like to be able to verify that the OpenMined library has properly implemented all of the core functionality by cross-checking it against the PyTorch testing suite.

## To do
Run one of the tests found [here](https://github.com/pytorch/pytorch/tree/master/test) replacing the PyTorch object with the corresponding OpenMined interface object (ex. `import torch` -> `import syft.interfaces.torch as torch`) and create an issue in the event that something doesn’t work (using the “pytorch” label).

`common.py` and `common_nn.py` should be the immediate priority, given that it defines a class `TestCase` which the other tests rely on. 

### Acceptance Criteria:
Either:
- File an issue with the "pytorch" label to signal which test you'll work on porting for OpenMined.
- Once you (or someone else) have successfully ported the test from PyTorch, file appropriate issues to signal areas where the testing suite fails. 

# Checklist (immediate priorities):
- [ ] Port `common.py` from [here](https://github.com/pytorch/pytorch/blob/master/test/common.py) to build a `TestCase` class for `syft.interfaces.torch`. Some things may not be directly ported (defining tensors on CPU/GPU), if this is the case, discuss this either commenting on this issue or in the Slack channel for #team_unity. 
- [ ] Port `common_nn.py` from [here](https://github.com/pytorch/pytorch/blob/master/test/common_nn.py).
- [ ] Port `test_torch.py` from [here](https://github.com/pytorch/pytorch/blob/master/test/test_torch.py).
- [ ] Port `test_nn.py` from [here](https://github.com/pytorch/pytorch/blob/master/test/test_nn.py).
